src/GSF.CarbonAware/src/Handlers/IEmissionsHandler.cs

```diff
@@ -1,7 +1,27 @@
+using GSF.CarbonAware.Models;
+
 namespace GSF.CarbonAware.Handlers
 {
     public interface IEmissionsHandler
     {
+        /// <summary>
+        /// Calculate the observed emission data by list of locations for a specified time period.
+        /// </summary>
+        /// <param name="locations">String array of named locations, must be non-empty (ex: ["eastus"])</param>
+        /// <param name="start">[Optional] Start time for the data query. (ex: 2022-03-01T15:30:00Z)</param>
+        /// <param name="end">[Optional] End time for the data query. (ex: 2022-03-01T18:30:00Z)</param>
+        /// <returns>Array of EmissionsData objects that contains the location, time and the rating in g/kWh</returns>
+        Task<IEnumerable<EmissionsData>> GetEmissionsDataAsync(string[] locations, DateTimeOffset? start = null, DateTimeOffset? end = null);
```

Comment:
It is overloaded to take a single location as well (this may have been an older commit you looked at). I think overall my thought is that enabling a single vs. multiple locations isn't a bit stretch and it is a small QOL improvement for users. In the newer commit, `GetEmissionsDataAsync` and `GetAverageCarbonIntensityAsync` are both overloaded for those improvements (single vs. multiple locations, and single vs. multiple time ranges)

Reply:
I am fine with it either way.  I am just questioning whether it is actually a QOL improvement or a cognitive/maintenance burden to have the option in the SDK without a clear use-case.  In all of the scenarios I could think of it felt easier to make the `_handler.SingleLocationFunction()` call within a loop that the developer controls than loop through the SDK responses. (With the exception of `Best` where we actually do work with the multi-location values)

Again, food for thought, not blocking.

Reply:
I see what mean now. Especially considering we do nothing fancy with the loop where a user may want to parallelize or have some more complex logic -> our solution doesn't add anything useful.